### PR TITLE
Code refactor (follow up to PR #1300)

### PR DESCRIPTION
**Pull request by amirmalka**
## Overview

Follow up for comments made in PR #1300 

List of changes:
- Replaced `fmt.Sprintf` with `strings.Join`
- Replaced `cautils.StringInSlice` with `slices.Contains`
- Renamed `KSResources` to `ExternalResources`
- Defined constant strings for field selectors logic
- Use `ScanInfo` in `GetResources` function

### CODE REVIEW

**Comment by dwertent** [comment]
We will first merge #1320

File: cmd/download/download.go [hunk 1]
```diff
@@ -55,7 +56,7 @@ func GetDownloadCmd(ks meta.IKubescape) *cobra.Command {
 			if len(args) < 1 {
 				return fmt.Errorf("policy type required, supported: %v", supported)
 			}
-			if cautils.StringInSlice(core.DownloadSupportCommands(), args[0]) == cautils.ValueNotFound {
+			if !slices.Contains(core.DownloadSupportCommands(), args[0]) {
```
[the remark this answers is not in the document]
Comment (matthyx) [reply cmd/download/download.go:59]: wait, wat?

[the remark this answers is not in the document]
Comment (matthyx) [reply cmd/download/download.go:59]: new slices package coming in 2 weeks (go 1.21) https://tip.golang.org/doc/go1.21#slices

[the remark this answers is not in the document]
Comment (amirmalka) [reply cmd/download/download.go:59]: > new slices package coming in 2 weeks (go 1.21) https://tip.golang.org/doc/go1.21#slices

I know... they're adding generics support, though not sure how much difference it's going to be like for strings

File: cmd/list/list.go [hunk 2]
```diff
@@ -43,7 +44,7 @@ func GetListCmd(ks meta.IKubescape) *cobra.Command {
 			if len(args) < 1 {
 				return fmt.Errorf("policy type requeued, supported: %s", supported)
 			}
-			if cautils.StringInSlice(core.ListSupportActions(), args[0]) == cautils.ValueNotFound {
+			if !slices.Contains(core.ListSupportActions(), args[0]) {
```
[the remark this answers is not in the document]
Comment (dwertent) [reply cmd/list/list.go:47]: /ask Why not use the built-in slices.Contains?

**Comment by dwertent** [comment]
/ask Why not use the built-in slices.Contains?

**Comment by dwertent** [comment]
/improve --pr_code_suggestions.extra_instructions="Add unit tests"

**Comment by dwertent** [comment]
/review --pr_reviewer.extra_instructions="focus on unit tests coverage"

**Comment by dwertent** [comment]
/describe --pr_description.publish_description_as_comment=true

**Review by dwertent — APPROVED** [review]

**Pull request merged**